projects/saturn/src/main/java/io/fairspace/saturn/services/workspaces/WorkspaceService.java

```diff
@@ -38,41 +38,51 @@ public WorkspaceService(Transactions tx, UserService userService) {
+                return new DAO(m)
+                        .list(Workspace.class).stream()
+                                .peek(ws -> {
+                                    var res = m.wrapAsResource(ws.getIri());
+                                    ws.setCanManage(userService.currentUser().isAdmin()
+                                            || user.hasProperty(FS.isManagerOf, res));
+                                    ws.setCanCollaborate(ws.isCanManage() || user.hasProperty(FS.isMemberOf, res));
+                                    var workspaceCollections = m.listSubjectsWithProperty(FS.ownedBy, res)
+                                            .filterKeep(r -> r.hasProperty(RDF.type, FS.Collection))
+                                            .toList();
+                                    var totalCollectionCount = workspaceCollections.size();
+                                    var nonDeletedCollectionCount = (int) workspaceCollections.stream()
+                                            .filter(collection -> !collection.hasProperty(FS.dateDeleted))
+                                            .count();
+                                    var memberCount = m.listSubjectsWithProperty(RDF.type, FS.User)
+                                            .filterKeep(u -> u.hasProperty(FS.isMemberOf, res))
+                                            .toList()
+                                            .size();
+                                    var managers = new DAO(m)
+                                            .list(User.class).stream()
+                                                    .filter(u -> m.wrapAsResource(u.getIri())
+                                                            .hasProperty(FS.isManagerOf, res))
+                                                    .collect(toList());
+                                    ws.setSummary(WorkspaceSummary.builder()
+                                            .totalCollectionCount(totalCollectionCount)
+                                            .nonDeletedCollectionCount(nonDeletedCollectionCount)
+                                            .memberCount(memberCount + managers.size())
+                                            .build());
+                                    ws.setManagers(managers);
+                                })
+                                .filter(ws -> userService.currentUser().isCanViewPublicMetadata()
+                                        || ws.isCanManage()
+                                        || ws.isCanCollaborate())
+                                .collect(toList());
+            } catch (Throwable e) {
+                log.error("+++++++++++++++++++++++++++++");
+                log.error("+++++++++++++++++++++++++++++");
+                log.error("+++++++++++++++++++++++++++++");
+                log.error("+++++++++++++++++++++++++++++");
+                log.error("+++++++++++++++++++++++++++++");
```

Comment:
Why such a strange log message?

Reply:
It shouldn't be there for sure as well as the added try-catch. I rolled it back. 

There was a local debugger's glitch - something did not work, but when I was trying to debug it got stuck.